src/scene_battle_rpg2k3.cpp

```diff
@@ -2083,6 +2089,65 @@ Scene_Battle_Rpg2k3::BattleActionReturn Scene_Battle_Rpg2k3::ProcessBattleAction
 		FaceTarget(*actor, *action->GetTarget());
 	}
 
+	if (action->GetCBAMovement() != lcf::rpg::BattlerAnimationItemSkill::Movement_none) {
+		SetBattleActionState(BattleActionState_MoveForward);
+	} else {
+		SetBattleActionState(BattleActionState_StartAlgo2);
+	}
+	return BattleActionReturn::eContinue;
+}
+
+Scene_Battle_Rpg2k3::BattleActionReturn Scene_Battle_Rpg2k3::ProcessBattleActionMoveForward(Game_BattleAlgorithm::AlgorithmBase* action) {
+	auto* source = action->GetSource();
```

Comment:
To make this more robust. We can use battle action substates. Make an initialization substate which resets the `cba_move_frame` variable. Then immediately go into the `Moving` substate which animates the move.

Basically, nothing outside of this function should be touching the `cba_move_...` variables. The function itself should be completely self contained.

Reply:
I'm probably missing something, but do battle action substates even exist? I only see scene action substates.

Reply:
Ahh you're right, battle action substates only exist in the 2k battle system.

In this case you could just create 2 battle action states. One which sets everything up and then immediately dispatches to the next which animates frame by frame.